static/app/views/organizationIntegrations/integrationExternalTeamMappings.tsx

```diff
@@ -83,83 +86,82 @@ class IntegrationExternalTeamMappings extends AsyncComponent<Props, State> {
     return externalTeamMappings.sort((a, b) => parseInt(a.id, 10) - parseInt(b.id, 10));
   }
 
-  sentryNamesMapper(teams: Team[]) {
-    return teams.map(({id, slug}) => ({id, name: slug}));
-  }
+  modalMappingKey = 'MODAL_RESULTS';
 
-  handleSubmit = (
-    data: Record<string, any>,
-    onSubmitSuccess: (data: Record<string, any>) => void,
-    onSubmitError: (error: any) => void,
-    _: React.FormEvent<Element>,
-    model: FormModel,
-    mapping?: ExternalActorMapping
-  ) => {
-    // We need to dynamically set the endpoint bc it requires the slug of the selected team in the form.
-    try {
-      const {organization} = this.props;
-      const {queryResults} = this.state;
-      const team = queryResults.find(item => item.id === data.teamId);
+  get dataEndpoint() {
+    const {organization} = this.props;
+    return `/organizations/${organization.slug}/teams/`;
+  }
 
-      if (!team) {
-        throw new Error('Cannot find team slug.');
-      }
+  getBaseFormEndpoint(mapping?: ExternalActorMapping) {
+    if (!mapping) {
+      return '';
+    }
+    const {organization} = this.props;
+    const {queryResults} = this.state;
+    const mappingResults =
+      queryResults[mapping.externalName] ?? queryResults[this.modalMappingKey];
+    const team = mappingResults?.find(item => item.id === mapping.teamId);
+    return `/teams/${organization.slug}/${team?.slug ?? ''}/external-teams/`;
+  }
 
-      const baseEndpoint = `/teams/${organization.slug}/${team.slug}/external-teams/`;
-      const apiEndpoint = mapping ? `${baseEndpoint}${mapping.id}/` : baseEndpoint;
-      const apiMethod = mapping ? 'PUT' : 'POST';
+  sentryNamesMapper(teams: Team[]) {
+    return teams.map(({id, slug}) => ({id, name: slug}));
+  }
 
-      model.setFormOptions({
-        onSubmitSuccess,
-        onSubmitError,
-        apiEndpoint,
-        apiMethod,
+  handleResults = (results, mappingKey?: string) => {
```

Comment:
This method runs for all the `onResults` hooks. It is using the mappingKey to store the results in state which we'll use later to create the endpoint we're submitting to.